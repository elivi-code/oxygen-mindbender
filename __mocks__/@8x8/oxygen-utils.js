export * from '@8x8/oxygen-utils';

const useFontSize = jest.fn(() => 16);

export { useFontSize };
