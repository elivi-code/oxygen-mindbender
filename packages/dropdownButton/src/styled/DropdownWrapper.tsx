import styled from 'styled-components';

import Button from '@8x8/oxygen-button';

export const StyledButton = styled(Button)<{ invertItems: boolean }>`
  display: inline-flex;
  align-items: center;
  ${({ invertItems }) => invertItems && 'flex-flow: row-reverse'}
`;

export const DropdownWrapper = styled.div`
  margin-top: ${({ theme }) => theme.dropdownOffset};

  padding: ${({ theme }) => theme.dropdownPadding};
  border: 1px solid ${({ theme }) => theme.dropdownBorderColor};
  border-radius: ${({ theme }) => theme.dropdownBorderRadius};

  box-shadow: ${({ theme }) => theme.dropdownShadowSize}
    ${({ theme }) => theme.dropdownShadowColor};
  background: ${({ theme }) => theme.dropdownBackgroundColor};

  overflow: hidden;
  z-index: ${({ theme }) => theme.dropdownZIndex};
`;

export default DropdownWrapper;
