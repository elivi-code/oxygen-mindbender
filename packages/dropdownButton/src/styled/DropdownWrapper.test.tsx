import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { dropdownButton as theme } from '@8x8/oxygen-constants';
import DropdownWrapper from './DropdownWrapper';

describe('<DropdownWrapper>', () => {
  let wrapper;

  function setup(props = { theme }) {
    return renderer.create(<DropdownWrapper {...props} />).toJSON();
  }

  it('should render correctly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
