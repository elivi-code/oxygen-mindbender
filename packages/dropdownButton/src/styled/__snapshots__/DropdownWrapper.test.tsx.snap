// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownWrapper> should render correctly 1`] = `
.c0 {
  margin-top: 0;
  padding: 4px 0;
  border: 1px solid #D1DBE8;
  border-radius: 0px;
  box-shadow: 0 2px 2px 0 #D1DBE8;
  background: #FFF;
  overflow: hidden;
  z-index: 3000;
}

<div
  className="c0"
/>
`;
